src/test/java/jenkins/plugins/logstash/LogstashWriterTest.java

```diff
@@ -266,14 +292,65 @@ public void writeBuildLogSuccess() throws Exception {
     // Verify results
     // No error output
     assertEquals("Results don't match", "", errorBuffer.toString());
-    verify(mockBuild).getLog(3);
+    String lines = JSONArray.fromObject(mockBuild.getLog(3)).toString();
+    verify(mockBuild, times(2)).getLog(3);
     verify(mockBuild).getCharset();
 
     verify(mockDao).buildPayload(eq(mockBuildData), eq("http://my-jenkins-url"), anyListOf(String.class));
-    verify(mockDao).push("{\"data\":{},\"message\":[\"test\"],\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
+    verify(mockDao).push("{\"data\":{},\"message\":" + lines + ",\"source\":\"jenkins\",\"source_host\":\"http://my-jenkins-url\",\"@version\":1}");
     verify(mockBuildData).updateResult();
   }
 
+  @Test
+  public void writeProcessedSuccess() throws Exception {
+    String goodMsg = "test message";
+    String ignoredMsg = "ignored input";
+    String scriptString =
+      "if (payload['message']) {\n" +
+      "  if (payload['message'][0] =~ /" + ignoredMsg + "/) {\n" +
+      "    payload.clear()\n" +
```

Comment:
To avoid assigning `null` I thought calling `clear()` makes sense. It would serve as a `Null` object. Do you like this or still want `payload = SKIP`?

Reply:
`payload.clear()` is fine